annotators/bot_emotion_classifier/server.py

```diff
@@ -0,0 +1,425 @@
+            type_num = get_dsript_type(sent, pos_type_expl)
+            bot_emotion = pos_reactions[type_num][emotion]
+        else:
+            bot_emotion = "neutral"
+    elif emotion in negative_emotions:
+        type_num = get_dsript_type(sent, neg_type_expl)
+        bot_emotion = neg_reactions[type_num][emotion]
+    else:
+        type_num = get_dsript_type(sent, pos_type_expl)
+        bot_emotion = pos_reactions[type_num][emotion]
+    return bot_emotion
+
+
+# the rate of mood decrease
+def get_dim_decay(default_dim, curr_dim):
+    p_dif = abs(default_dim - curr_dim)
+    if p_dif == 0:
+        p_dif = 0.00001
+    decay = 1 / p_dif * 0.01
+    if decay > p_dif:
+        decay = p_dif
+    if default_dim < curr_dim:
+        decay = -decay
+    return decay
+
+
+def get_decay(default_mood, curr_mood):
+    decay = []
+    for i in range(len(default_mood)):
+        decay.append(get_dim_decay(default_mood[i], curr_mood[i]))
+    return decay
+
+
+# comparison of old and new mood
+def check_same_mood(curr_mood, new_mood):
+    print(curr_mood, new_mood)
+    for i in range(len(curr_mood)):
+        if curr_mood[i] * new_mood[i] < 0:
+            return False
+    return True
+
+
+# new mood calculation using difference btw current mood and current emotion
+def get_new_mood(default_mood, curr_mood, bot_emotion):
+    if check_same_mood(pad_emotions[bot_emotion], curr_mood):
+        decay = [0, 0, 0]
+    else:
+        decay = get_decay(default_mood, curr_mood)
+    vec = [pad_emotions[bot_emotion][i] - curr_mood[i] for i in range(len(curr_mood))]
+    new_mood = [0.5 * curr_mood[i] + 0.5 * vec[i] + decay[i] for i in range(len(curr_mood))]
+    new_mood_reg = [1 if dim > 1 else dim for dim in new_mood]
+    new_mood_reg = [-1 if dim < -1 else dim for dim in new_mood_reg]
+
+    # dim_symbols = [str(int(dim / abs(dim))) if dim != 0 else '-1' for dim in new_mood_reg]
+    # octant = ''.join(dim_symbols)
+    # print('New mood:', pad_moods[octant])
+    return new_mood_reg
+
+
+def get_mood_label(bot_mood):
+    octant = ""
+    for dim in bot_mood:
+        if dim == 0:
+            octant += "0"
+        elif dim > 0:
+            octant += "1"
+        else:
+            octant += "-1"
+
+    return pad_moods[octant]
+
+
+@app.route("/model", methods=["POST"])
+def respond():
+    sentences = request.json.get("sentences", [])
+    annotated_utterances = request.json.get("annotated_utterances", [])
+    bot_moods = request.json.get("bot_mood", [])
+
+    for i in range(len(sentences)):
+        sentence = sentences[i]
+        annotated_utterance = annotated_utterances[i]
+        bot_mood = bot_moods[i]
+
+        user_emotion = get_emotions(annotated_utterance, probs=False)[0]
+        sentiment = get_sentiment(annotated_utterance, probs=False)[0]
+
+        print("SENTENCE:", sentence)
+        print("USER EMOTION: ", user_emotion)
+        print("SENTIMENT: ", sentiment)
+        print("OLD BOT MOOD: ", bot_mood)
```

Comment:
all prints should be deleted (or changed to logger) -- they will not be shown because of logger

Reply:
deleted them (I already had the same info in logger)